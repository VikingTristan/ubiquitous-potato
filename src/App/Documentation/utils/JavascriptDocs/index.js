import React from "react";
import { Link } from "react-router-dom";

import { DgScript } from "@docutils";

const OpenDocs = ({ componentName }) => (
    <>
        <tr>
            <td scope="row"><DgScript component={componentName} func="open" params={[`<${componentName.toLowerCase()}-id>`]} /></td>
            <td>Opens the given {componentName.toLowerCase()}</td>
        </tr>
    </>
);

const CloseDocs = ({ componentName }) => (
    <>
        <tr>
            <td scope="row"><DgScript component={componentName} func="close" params={[`<${componentName.toLowerCase()}-id>`]} /></td>
            <td>Closes the given {componentName.toLowerCase()}</td>
        </tr>
    </>
);

const JavascriptDocs = ({ componentName, open, close, others }) => (
    <>
        <table className="table table-striped">
            <thead>
                <tr>
                    <th scope="col">Function</th>
                    <th scope="col">Purpose</th>
                </tr>
            </thead>
            <tbody>
                <tr>
                    <td scope="row"><DgScript component={componentName} func="init" /></td>
                    <td><Link to="/docs/getting-started/javascript">Initializes</Link> {componentName.toLowerCase()}</td>
                </tr>
                {open ? <OpenDocs componentName={componentName} /> : null}
                {close ? <CloseDocs componentName={componentName} /> : null}
                {others && others.map((DocComponent, i) => <DocComponent key={i} componentName={componentName} />)}
            </tbody>
        </table>
    </>
);

export default JavascriptDocs;

/* For testing */
export { OpenDocs, CloseDocs };
