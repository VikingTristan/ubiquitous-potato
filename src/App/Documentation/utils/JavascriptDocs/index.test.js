import React from "react";
// import { Link } from "react-router-dom";
import { shallow, mount } from "enzyme";

import JavascriptDocs from "./index";

jest.mock("react-router-dom");

describe("Utilities: JavascriptDocs", () => {
    it("is defined", () => {
        expect(JavascriptDocs).toBeDefined();
    });

    it("renders only init method", () => {
        const wrapper = shallow(<JavascriptDocs componentName="test" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains("open")).toBeFalsy();
        expect(wrapper.contains("close")).toBeFalsy();
    });

    it("renders init and open methods", () => {
        const wrapper = mount(
            <JavascriptDocs componentName="test" open />
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains("open")).toBeTruthy();
        expect(wrapper.contains("close")).toBeFalsy();
    });

    it("renders init and close methods", () => {
        const wrapper = mount(
            <JavascriptDocs componentName="test" close />
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains("open")).toBeFalsy();
        expect(wrapper.contains("close")).toBeTruthy();
    });

    it("renders init, open and close methods", () => {
        const wrapper = mount(
            <JavascriptDocs componentName="test" open close />
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains("open")).toBeTruthy();
        expect(wrapper.contains("close")).toBeTruthy();
    });
});

