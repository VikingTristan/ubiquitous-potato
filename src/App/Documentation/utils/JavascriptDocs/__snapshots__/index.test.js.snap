// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utilities: JavascriptDocs renders init and close methods 1`] = `
<JavascriptDocs
  close={true}
  componentName="test"
>
  <table
    className="table table-striped"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Function
        </th>
        <th
          scope="col"
        >
          Purpose
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          scope="row"
        >
          <DgScript
            component="test"
            func="init"
          >
            <code>
              <span>
                dg
              </span>
              <span
                className="token punctuation"
              >
                .
              </span>
              <span>
                test
              </span>
              <span
                className="token punctuation"
              >
                .
              </span>
              <span
                className="token function"
              >
                init
              </span>
              <span
                className="token punctuation"
              >
                (
              </span>
              <span
                className="token punctuation"
              >
                );
              </span>
            </code>
          </DgScript>
        </td>
        <td>
          <Link
            to="/docs/getting-started/javascript"
          />
           
          test
        </td>
      </tr>
      <CloseDocs
        componentName="test"
      >
        <tr>
          <td
            scope="row"
          >
            <DgScript
              component="test"
              func="close"
              params={
                Array [
                  "<test-id>",
                ]
              }
            >
              <code>
                <span>
                  dg
                </span>
                <span
                  className="token punctuation"
                >
                  .
                </span>
                <span>
                  test
                </span>
                <span
                  className="token punctuation"
                >
                  .
                </span>
                <span
                  className="token function"
                >
                  close
                </span>
                <span
                  className="token punctuation"
                >
                  (
                </span>
                <span
                  className={null}
                  key="0"
                >
                  &lt;test-id&gt;
                </span>
                <span
                  className="token punctuation"
                >
                  );
                </span>
              </code>
            </DgScript>
          </td>
          <td>
            Closes the given 
            test
          </td>
        </tr>
      </CloseDocs>
    </tbody>
  </table>
</JavascriptDocs>
`;

exports[`Utilities: JavascriptDocs renders init and open methods 1`] = `
<JavascriptDocs
  componentName="test"
  open={true}
>
  <table
    className="table table-striped"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Function
        </th>
        <th
          scope="col"
        >
          Purpose
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          scope="row"
        >
          <DgScript
            component="test"
            func="init"
          >
            <code>
              <span>
                dg
              </span>
              <span
                className="token punctuation"
              >
                .
              </span>
              <span>
                test
              </span>
              <span
                className="token punctuation"
              >
                .
              </span>
              <span
                className="token function"
              >
                init
              </span>
              <span
                className="token punctuation"
              >
                (
              </span>
              <span
                className="token punctuation"
              >
                );
              </span>
            </code>
          </DgScript>
        </td>
        <td>
          <Link
            to="/docs/getting-started/javascript"
          />
           
          test
        </td>
      </tr>
      <OpenDocs
        componentName="test"
      >
        <tr>
          <td
            scope="row"
          >
            <DgScript
              component="test"
              func="open"
              params={
                Array [
                  "<test-id>",
                ]
              }
            >
              <code>
                <span>
                  dg
                </span>
                <span
                  className="token punctuation"
                >
                  .
                </span>
                <span>
                  test
                </span>
                <span
                  className="token punctuation"
                >
                  .
                </span>
                <span
                  className="token function"
                >
                  open
                </span>
                <span
                  className="token punctuation"
                >
                  (
                </span>
                <span
                  className={null}
                  key="0"
                >
                  &lt;test-id&gt;
                </span>
                <span
                  className="token punctuation"
                >
                  );
                </span>
              </code>
            </DgScript>
          </td>
          <td>
            Opens the given 
            test
          </td>
        </tr>
      </OpenDocs>
    </tbody>
  </table>
</JavascriptDocs>
`;

exports[`Utilities: JavascriptDocs renders init, open and close methods 1`] = `
<JavascriptDocs
  close={true}
  componentName="test"
  open={true}
>
  <table
    className="table table-striped"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Function
        </th>
        <th
          scope="col"
        >
          Purpose
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          scope="row"
        >
          <DgScript
            component="test"
            func="init"
          >
            <code>
              <span>
                dg
              </span>
              <span
                className="token punctuation"
              >
                .
              </span>
              <span>
                test
              </span>
              <span
                className="token punctuation"
              >
                .
              </span>
              <span
                className="token function"
              >
                init
              </span>
              <span
                className="token punctuation"
              >
                (
              </span>
              <span
                className="token punctuation"
              >
                );
              </span>
            </code>
          </DgScript>
        </td>
        <td>
          <Link
            to="/docs/getting-started/javascript"
          />
           
          test
        </td>
      </tr>
      <OpenDocs
        componentName="test"
      >
        <tr>
          <td
            scope="row"
          >
            <DgScript
              component="test"
              func="open"
              params={
                Array [
                  "<test-id>",
                ]
              }
            >
              <code>
                <span>
                  dg
                </span>
                <span
                  className="token punctuation"
                >
                  .
                </span>
                <span>
                  test
                </span>
                <span
                  className="token punctuation"
                >
                  .
                </span>
                <span
                  className="token function"
                >
                  open
                </span>
                <span
                  className="token punctuation"
                >
                  (
                </span>
                <span
                  className={null}
                  key="0"
                >
                  &lt;test-id&gt;
                </span>
                <span
                  className="token punctuation"
                >
                  );
                </span>
              </code>
            </DgScript>
          </td>
          <td>
            Opens the given 
            test
          </td>
        </tr>
      </OpenDocs>
      <CloseDocs
        componentName="test"
      >
        <tr>
          <td
            scope="row"
          >
            <DgScript
              component="test"
              func="close"
              params={
                Array [
                  "<test-id>",
                ]
              }
            >
              <code>
                <span>
                  dg
                </span>
                <span
                  className="token punctuation"
                >
                  .
                </span>
                <span>
                  test
                </span>
                <span
                  className="token punctuation"
                >
                  .
                </span>
                <span
                  className="token function"
                >
                  close
                </span>
                <span
                  className="token punctuation"
                >
                  (
                </span>
                <span
                  className={null}
                  key="0"
                >
                  &lt;test-id&gt;
                </span>
                <span
                  className="token punctuation"
                >
                  );
                </span>
              </code>
            </DgScript>
          </td>
          <td>
            Closes the given 
            test
          </td>
        </tr>
      </CloseDocs>
    </tbody>
  </table>
</JavascriptDocs>
`;

exports[`Utilities: JavascriptDocs renders only init method 1`] = `
<Fragment>
  <table
    className="table table-striped"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Function
        </th>
        <th
          scope="col"
        >
          Purpose
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          scope="row"
        >
          <DgScript
            component="test"
            func="init"
          />
        </td>
        <td>
          <Link
            to="/docs/getting-started/javascript"
          >
            Initializes
          </Link>
           
          test
        </td>
      </tr>
    </tbody>
  </table>
</Fragment>
`;
